datahub-web-react/src/app/entity/shared/tabs/Dataset/Validations/contract/builder/DataContractBuilder.tsx

```diff
@@ -110,6 +110,21 @@ export const DataContractBuilder = ({ entityUrn, initialState, onSubmit, onCance
 
     const hasAssertions = freshnessAssertions.length || schemaAssertions.length || dataQualityAssertions.length;
 
+    const onSelectFreshnessOrSchemaAssertion = (assertionUrn: string, type: string) => {
+        const selected = builderState[type]?.assertionUrn === assertionUrn;
```

Comment:
does this mean user can only select one of each type?

Reply:
the logic is reused.. it will check for particular assertion if and add only one assertion for freshness and Schema.